planetscale/dbutil/dbutil.go

```diff
@@ -54,7 +60,11 @@ func Dial(ctx context.Context, cfg *DialConfig) (*sql.DB, error) {
 	if err != nil {
 		return nil, err
 	}
-	mysqlCfg := mysql.NewConfig()
+
+	mysqlCfg := cfg.MySQLConfig
```

Comment:
Should we copy the configuration? To avoid modifying the user's configuration.

```suggestion
	mysqlCfg := *cfg.MySQLConfig
```

Reply:
For sure 👍🏼 Hopefully as long dbutil doesn't ever need to modify the non-shallow fields defined here: https://github.com/go-sql-driver/mysql/blob/b36cd86ebcb680d317f962679780f5877a0b91e1/dsn.go#L36 in which case a `*cfg.MySQLConfig` may not be enough. 

So far it only modifies strings so I'll commit the suggestion above. Thanks!

Reply:
@fatih  Actually I just realize this change might end up causing a panic. Pushing a new commit

Reply:
Nice, indeed it would panic! Thanks for the commit :)